server/plugins/api/GenericNotifier.ts

```diff
@@ -0,0 +1,7 @@
+import { GenericConfig } from "./GenericConfig";
+import { APIReturn } from "./APIReturn";
+
+export interface GenericNotifier {
+	sendMessage<T extends GenericConfig>(config: T): APIReturn;
```

Comment:
We can get rid of the `GenericConfig` interface (thereby simplifying code) by changing the method definition to just include `message`.

Also, I think you'll get an error later on when plugins try to implement `GenericNotifier` since they have to support any `T` in their configuration (since the generic is on the function instead of on the interface). I think this is correct:

```typescript
export interface GenericNotifier<T> {
	async sendMessage(message: String, config: T): Promise<APIReturn>;
}
```

Note these methods should also be `async`.
If the above doesn't work then maybe it should just be: `sendMessage(message: String, config: any)`.
Getting the config type to typecheck will be difficult, and probably for a later PR.